variant.d

```diff
@@ -234,6 +234,12 @@ struct Variant{
 		assert(occupies == Occupies.vars);
 	}
 
+	this()(typeof(null), Type type = null){ // templated because of DMD bug
```

Comment:
I guess it would be better if the type was set to Type.get!(typeof(null))().
Also, I don't think you need the Type argument. (I think the only place where Variant(null) is used is as a dummy value in order to initialize a built-in array in order to force reference semantics -- i.e. at the moment it shouldn't matter, but in the future somebody might expect Variant(null) to be the constant 'null' value.)

I.e., what you did works, but this is better:
```
	this()(typeof(null)){ // templated because of DMD bug
		this.type=Type.get!(typeof(null))();
		this.vars=null;
	}
```